pkg/machineoutput/types.go

```diff
@@ -0,0 +1,59 @@
+package machineoutput
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/openshift/odo/pkg/log"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Kind is what kind we should use in the machine readable output
+const Kind = "Error"
+
+// APIVersion is the current API version we are using
+const APIVersion = "odo.openshift.io/v1alpha1"
+
+// Error for machine readable output error messages
+type Error struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Message           string `json:"message"`
+}
+
+// Success same as above, but copy-and-pasted just in case
+// we change the output in the future
+type Success struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Message           string `json:"message"`
```

Comment:
Why are we not using the already existing machine readable spec for project here?

Reply:
What do you mean? From `project/project.go`? This is a "generic" success output

Reply:
I mean shouldn't we output other data like `Spec` and `Active`? Or am I missing something?

Reply:
Nope, we removed `Active` and `Spec` would not be needed.

Reply:
@mik-dass  is right if you look at other resources that we are returning as JSON the data is always in `Spec` field. We want our output to resemble CRDs, so we are using Kubernetes convention here.

It should look something like this:
``` go
type Success struct {
	metav1.TypeMeta   `json:",inline"`
	metav1.ObjectMeta `json:"metadata,omitempty"`
        Spec SuccessSpec `json:"spec,omitempty"`
}

type SuccessSpec struct {
	Message           string `json:"message"`
}
```

Similarly for `Error`